### DISCUSSION TITLE
Changing canvas.freeDrawingCursor requires mouse move to update changes.

### DISCUSSION THREAD

**Discussion by AlvesJorge** [fabricjs/fabric.js#8645] [Q&A] [answered]
Using this function I found [here](https://codepen.io/curtisj44/pen/yGxJNX) to change the cursor of my fabric.js canvas, on zoom (change size) and on keyboard event (change color of the cursor).
This is the function:
```
function getDrawCursor(brushColor, zoom = 1) {
	console.log(zoom)
	const circle = "
		<svg
			height="${strokeWidth * zoom}"
			width="${strokeWidth * zoom}"
			fill="${brushColor}"
			viewBox="0 0 ${strokeWidth * zoom * 2} ${strokeWidth * zoom * 2}"
			xmlns="http://www.w3.org/2000/svg"
		>
			<circle
				cx="50%"
				cy="50%"
				r="${strokeWidth * zoom}" 
			/>
		</svg>
	";
        return "data:image/svg+xml;base64,${window.btoa(circle)}";
};
```

I initialize canvas with the cursor like this:
```
canvasName = new fabric.Canvas(canvasElem, {
		isDrawingMode: true,
		...
		freeDrawingCursor: "url(${getDrawCursor(redBrushColor)}) ${strokeWidth / 2} ${strokeWidth / 2}, crosshair",
	});`
```
Then I change the cursor color using the keyboard like this:

```
window.addEventListener("keydown", (e) => {
  switch (e.key) {
    case "c":
	if(canvas.freeDrawingBrush.color = "red"){
	canvas.freeDrawingBrush.color = "green"
	canvas.freeDrawingCursor = "url(${getDrawCursor(greenBrushColor)}) ${strokeWidth / 2} ${strokeWidth / 2}, crosshair"
	}		
}
})
```
This code is edited down but this is in summary all the key parts. The cursor updates and the change is as intended but only after I move the mouse. If I press "c" and not move the mouse, absolutely nothing happens. 
It could be related to [object caching](http://fabricjs.com/fabric-gotchas#:~:text=Object%20does%20not%20update%20after%20changing%20property%20%2D%20objectCaching) but I'm not sure how to use this information to fix my issue.
Any ideas on how to deal with this? Thank you :pray:

**Comment by ShaMan123** [comment, marked as answer]
First of, good to know it is possible to produce a cursor with svg. You can even use fabric Objects toSVG to do that. Very cool.
Second, you are correct. `setCursor` is called on mouse move. That sounds wrong.
https://github.com/fabricjs/fabric.js/search?q=setCursor
So simply call `setCursor`

**Reply by ShaMan123** [reply]
> It could be related to [object caching](http://fabricjs.com/fabric-gotchas#:~:text=Object%20does%20not%20update%20after%20changing%20property%20%2D%20objectCaching) but I'm not sure how to use this information to fix my issue.

Unrelated, fabric uses the cursor from the window

**Reply by AlvesJorge** [reply]
Using setCursor works perfectly and seems to be the right way to do it, thank you ShaMan123!

**Reply by ShaMan123** [reply]
#8476 is a rewrite to the brush API. I have changed how the cursor is set, thanks as well!
